system-addon/lib/SectionsManager.jsm

```diff
@@ -39,13 +40,19 @@ const BUILT_IN_SECTIONS = {
 
 const SectionsManager = {
   ACTIONS_TO_PROXY: ["SYSTEM_TICK", "NEW_TAB_LOAD"],
+  CONTEXT_MENU_PREFS: {"SaveToPocket": "extensions.pocket.enabled"},
   initialized: false,
   sections: new Map(),
   init(prefs = {}) {
     for (const feedPrefName of Object.keys(BUILT_IN_SECTIONS)) {
       const optionsPrefName = `${feedPrefName}.options`;
       this.addBuiltInSection(feedPrefName, prefs[optionsPrefName]);
     }
+
+    this.prefsObserver = () => this.updateSections();
```

Comment:
Just add an `observe` method to the object and then you can call `addObserver(pref, this)`

https://github.com/mozilla/activity-stream/blob/17e5e5bd4bd5917cda4114840feb375d14db2b9f/system-addon/lib/ActivityStream.jsm#L278-L288